### PR TITLE
Develop to master

### PR DESCRIPTION
**Pull request by tmisawa**
**bug fix**
- specify ip in calham.c and calham_real.c as shared #3 
- fix bug in calculating 2-body Green functions in Lanczos with complex variational parameters
- bug fix in fourier2k [density-density correlations]
- bug fix for interall in UHF calculations

**new functions**
- adding ctest
- adding GHActions
- adding blocked pfaffian calculations

### CODE REVIEW

**Comment by tmisawa** [comment]
@k-yoshimi @xrq-phys @yomichi @k-ido 
Now, PFAFFIAN_BLOCKED is "ON" in 
intel.cmake, aocc.cmake, and fugaku.cmake. 
PFAFFIAN_BLOCKED is an advanced option and
it is better to set  PFAFFIAN_BLOCKED=OFF in the default setting.
How do you think?

In addition, in obcx and our clusters, I found that 
`-DCMAKE_CXX_FLAGS="-std=c++11"` or `-std=c++11`
is necessary for compiling the blocked pfaffian submodule.
I think this is just because the GCC in the clusters is old.
However, for usability, it is better to add the above options in the config files.

**Comment by xrq-phys** [comment]
@tmisawa `PFAFFIAN_BLOCKED` has a default value `OFF`.
Forcefully setting it to be `OFF` in config file would result in commands like `cmake .. -DPFAFFIAN_BLOCKED=ON` being ineffective.

If keeping default `OFF` is what should be expected, we should delete those lines instead of setting it `OFF`. Otherwise, persons who wants it be `ON` shall suffer from inconvenience.

File: config/aocc.cmake [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+set(CMAKE_Fortran_COMPILER "flang" CACHE STRING "" FORCE)
+set(CMAKE_Fortran_FLAGS_RELEASE "-O3 -DNDEBUG -DHAVE_SSE2" CACHE STRING "" FORCE)
+
+# for AOCL
+set(BLA_VENDOR "FLAME" CACHE STRING "" FORCE)
+
+# for BLIS & block pfaffian
+set(BLIS_ARTIFACT_CONFIG "amd64")
+set(PFAFFIAN_BLOCKED OFF)
```
Comment (xrq-phys) [inline config/aocc.cmake:17]: Delete this line instead of setting it `OFF`.

File: config/fugaku.cmake [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+set(OpenMP_Fortran_FLAGS "-Kopenmp" CACHE STRING "" FORCE)
+set(CMAKE_EXE_LINKER_FLAGS "-fopenmp" CACHE STRING "" FORCE)
+
+# for SSL2
+set(BLAS_LIBRARIES "-SSL2" CACHE STRING "" FORCE)
+set(LAPACK_LIBRARIES ${BLAS_LIBRARIES} CACHE STRING "" FORCE)
+
+# for BLIS & block pfaffian
+set(BLIS_ARTIFACT_CONFIG "a64fx")
+set(PFAFFIAN_BLOCKED OFF)
```
Comment (xrq-phys) [inline config/fugaku.cmake:20]: Delete this line instead of setting it `OFF`.

File: config/intel.cmake [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
 
 set(CMAKE_Fortran_COMPILER "ifort" CACHE STRING "" FORCE)
 set(CMAKE_Fortran_FLAGS_RELEASE "-O3 -DNDEBUG -DHAVE_SSE2" CACHE STRING "" FORCE)
 
 # for Intel MKL
 set(BLA_VENDOR "Intel10_64lp" CACHE STRING "" FORCE)
+
+# for BLIS & block pfaffian
+set(BLIS_ARTIFACT_CONFIG "intel64")
+set(PFAFFIAN_BLOCKED OFF)
```
Comment (xrq-phys) [inline config/intel.cmake:17]: Delete this line instead of setting it `OFF`.

**Comment by xrq-phys** [comment]
@tmisawa Thanks for confirming.

**Comment by tmisawa** [comment]
@xrq-phys 
Thank you for your comments!
I commented out `PFAFFIAN_BLOCKED OFF`.

**Pull request merged**